python/cudf/cudf/core/column/column.py

```diff
@@ -1195,6 +1195,19 @@ def as_column(arbitrary, nan_as_null=True, dtype=None, name=None):
                 mask = libcudf.unaryops.nans_to_nulls(data)
                 data = data.set_mask(mask)
 
+        elif data.dtype.kind == "M":
+            null = cudf.Series(np.array([None]))
```

Comment:
directly construct the column here instead of constructing a Series and reaching in for the column

Reply:
Can also construct the column with the correct dtype directly instead of needing to typecast later

Reply:
> Can also construct the column with the correct dtype directly instead of needing to typecast later

Directly passing dtype changes the value from `None` to `NaT`,  so can't do it.

Reply:
you should be able to do:
```cudf.Series(pa.array([None], type=pa.timestamp64('ms'))``` for example. You're effectively forcing a `NaN` --> `null` conversion here.

Reply:
I found other way, `null = as_column(np.array([None]))` and then `null.as_datetime_column(array.dtype)`

Reply:
`np.array([None])` I believe will create a numpy array with one element of `NaN` of type float64 or a numpy array with one element `None` of Python objects. Either way that's not what you want as it will then run a tiny kernel to convert it to a `null` and then run an additional tiny kernel to handle the typecasting.

Reply:
Actually, we build `column_empty_like` for this very reason, should use that. 👍